cardano-cli/src/Cardano/CLI/Shelley/Run/Pretty.hs

```diff
@@ -1,107 +1,180 @@
+import           Ouroboros.Consensus.Shelley.Protocol.Crypto (StandardCrypto)
+import           Shelley.Spec.Ledger.API (Addr (..), TxOut (TxOut))
 import qualified Shelley.Spec.Ledger.API as Shelley
 
-prettyTx :: TxBody era -> LByteString
-prettyTx body0 =
-  encodePretty' defConfig{confCompare = compare} $
-  object $
-  case body0 of
-    ByronTxBody tx -> ["era" .= ("Byron" :: Text), "tx" .= tx]
-    ShelleyTxBody ShelleyBasedEraShelley body aux ->
-      [ "era" .= ("Shelley" :: Text)
-      , "inputs" .= _inputs
-      , "outputs" .= _outputs
-      , "certificates" .= fmap textShow _certs
-      , "withdrawals" .= withdrawals
-      , "fee" .= _txfee
-      , "timetolive" .= _ttl
-      , "update" .= fmap textShow _txUpdate
-      , "metadata_hash" .= fmap textShow _mdHash
-      , "auxiliary_data" .= fmap textShow aux
-      ]
-      where
-        Shelley.TxBody
-          { _inputs
-          , _outputs
-          , _certs
-          , _wdrls
-          , _txfee
-          , _ttl
-          , _txUpdate
-          , _mdHash
-          } =
-            body
-        Shelley.Wdrl withdrawals = _wdrls
-    ShelleyTxBody ShelleyBasedEraAllegra body aux ->
-      [ "era" .= ("Allegra" :: Text)
-      , "inputs" .= inputs
-      , "outputs" .= outputs
-      , "certificates" .= fmap textShow certificates
-      , "withdrawals" .= withdrawals
-      , "fee" .= txfee
-      , "validity_interval" .= prettyValidityInterval validity
-      , "update" .= fmap textShow update
-      , "auxiliary_data_hash" .= fmap textShow adHash
-      , "mint" .= mint
-      , "auxiliary_data" .= fmap textShow aux
-      ]
-      where
-        ShelleyMA.TxBody
-          inputs
-          outputs
-          certificates
-          (Shelley.Wdrl withdrawals)
-          txfee
-          validity
-          update
-          adHash
-          mint =
-            body
-    ShelleyTxBody ShelleyBasedEraMary body aux ->
-      [ "era" .= ("Mary" :: Text)
-      , "inputs" .= inputs
-      , "outputs" .= outputs
-      , "certificates" .= fmap textShow certificates
-      , "withdrawals" .= withdrawals
-      , "fee" .= txfee
-      , "validity_interval" .= prettyValidityInterval validity
-      , "update" .= fmap textShow update
-      , "auxiliary_data_hash" .= fmap textShow adHash
-      , "mint" .= mint
-      , "auxiliary_data" .= fmap textShow aux
-      ]
-      where
-        ShelleyMA.TxBody
-          inputs
-          outputs
-          certificates
-          (Shelley.Wdrl withdrawals)
-          txfee
-          validity
-          update
-          adHash
-          mint =
-            body
+friendlyTxBodyBS :: Api.TxBody era -> ByteString
+friendlyTxBodyBS =
+  encodePretty (setConfCompare compare defConfig) . friendlyTxBody
+
+friendlyTxBody :: Api.TxBody era -> Value
+friendlyTxBody = \case
+  ByronTxBody tx ->
+    _Object (HashMap.insert "era" "Byron") $ friendlyTxBodyByron tx
```

Comment:
I think a more concise was of doing this would be:
```
mergeEra :: CardanoEra era -> JSON.Value -> JSON.Value
mergeEra era (JSON.Object obj) =
  case toJSON era of 
    JSON.Object e -> JSON.Object $ e <> obj
    _ -> JSON.Object obj
mergeEra _   _  = JSON.Null
```
And defining a suitable `ToJSON` instance for `CardanoEra era` i.e
```
toJSON ByronEra = object [ "era" .= Aeson.String "ByronEra" ]
```
etc

Have a look at https://github.com/input-output-hk/cardano-node/pull/2465/files#diff-cc9b567f3acd813c81e105c074ebd783719c74364c88a78c3c22638322354f6dR789
You will notice that we have access to `CardanoEra era` (`_era`). We can use this to simplify the pattern matching.

Reply:
The user-friendly name for the Byron era is "Byron", not "ByronEra".

I'm not sure this is concise, but I rewrote it a bit differently, please see.

Reply:
It must be done separately. I've created #2533.